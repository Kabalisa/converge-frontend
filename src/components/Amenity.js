import React from 'react';
import PropTypes from 'prop-types';

class Amenity extends React.Component {
  static propTypes = {
    amenity: PropTypes.shape({
      name: PropTypes.string.isRequired,
      rooms: PropTypes.number.isRequired,
    }).isRequired,
  }

  render() {
    const { name, rooms } = this.props.amenity;
    return (
      <tr>
        <td>{name}</td>
        <td>{rooms}</td>
        <td>Edit &emsp; Delete</td>
      </tr>
    );
  }
}

export default Amenity;
