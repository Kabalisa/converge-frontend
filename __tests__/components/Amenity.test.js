import React from 'react';
import { shallow } from 'enzyme';

import Amenity from '../../src/components/Amenity';

describe('Tests for Amenity Component', () => {
  const amenity = {
    name: 'AmenityName',
    rooms: 8,
  };

  const shalloWrapper = shallow(<Amenity amenity={amenity} />);

  it('renders correctly from memory', () => {
    expect(shalloWrapper).toMatchSnapshot();
  });
});
