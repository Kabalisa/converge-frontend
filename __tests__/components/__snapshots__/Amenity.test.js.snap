// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Amenity Component renders correctly from memory 1`] = `
<tr>
  <td>
    AmenityName
  </td>
  <td>
    8
  </td>
  <td>
    Edit â€ƒ Delete
  </td>
</tr>
`;
